Kuzzle.Tests/API/Controllers/IndexControllerTest.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using KuzzleSdk.API.Controllers;
+using KuzzleSdk.API.DataObjects;
+using KuzzleSdk.API.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Kuzzle.Tests.API.Controllers {
+  public class IndexControllerTest {
+    private readonly IndexController _indexController;
+    private readonly KuzzleApiMock _api;
+
+    public IndexControllerTest() {
+      _api = new KuzzleApiMock();
+      _indexController = new IndexController(_api.MockedObject);
+    }
+
+    [Fact]
+    public async void CreateAsyncTest() {
+      var expected = JObject.Parse(@"
+      {
+				acknowledged: true, 
+				shards_acknowledged: true
+			} ");
+
+      _api.SetResult(new JObject { { "result", expected } });
+
+      JObject result = await _indexController.CreateAsync("foo");
+      _api.Verify(new JObject {
+        { "controller", "index" },
+        { "action", "create" },
+        { "index", "foo" }
+      });
+
+			Assert.Equal(expected, result);
+		}
+
+    [Fact]
+    public async void DeleteAsyncTest() {
+      _api.SetResult(@"{result: {acknowledged: true} }");
+      await _indexController.DeleteAsync("foo");
+
+      _api.Verify(new JObject {
+        { "controller", "index" },
+        { "action", "delete" },
+        { "index", "foo" }
+      });
+		}
+
+    [Fact]
+    public async void ExistsAsyncTest() {
+      _api.SetResult(@"{result: true}");
```

Comment:
Unit tests should test all possible outcomes (use a theory instead of a fact). This should be fixed for all comparable tests.